### DISCUSSION TITLE
Bills and budget extra costs

### DISCUSSION THREAD

**Discussion by Solverz-0** [firefly-iii/firefly-iii#6538] [Firefly III] [answered]
Hope someone can clear this up for me!

If there is a repeat monthly bill for 100 but on month 2 this bill is 150 due to some extra fees. This extra 50 is a one off, so would it be best to link the 150 transaction to the bill or split the transaction into 100 and 50 and only link the 100 to the bill and do not link the 50?
If the latter what would I do with the 50? Leave it and do not link it to anything or create a one off bill and link it to that?

Also what about budgets, this extra 50 would put my bills budget overbudget for the month as it was unexpected, would I move money from savings asset account to current asset account and increase the budget for bills for that month to compensate for this or leave the budget alone and just move the money so the current asset account is not negative?

Hope this makes sense, just getting my understanding of how things should flow :)

**Comment by JC5** [comment, marked as answer]
Of course, in the end this is all up to you, but here is what I do:

- I link the whole transaction to the bill. No splitting. It's not a problem and it will show that the amount was higher than expected.
- I transfer extra money (50) to the asset account so I'm not negative.
- I don't make the budget higher, so it remains visible that I went over budget.

**Reply by Solverz-0** [reply]
Thank you for this, it has cleared things up for me.

I think it helps a lot to see how you handle some use cases with examples like this :)
I also would like to suggest maybe a "use case examples" section in the documentation?

**Reply by Solverz-0** [reply]
I just found out that if I overay for a bill for a month, then the `Bills to pay` section will not be correct as it will take the ovepayment into account in the calculation, so if I overpaid `bill A` by `$100` but still owed `bill B` `$100`, then the `Bills to pay` will show `$0` which is incorrect, any ideas? :)

**Reply by JC5** [reply]
That's not how that calculation works, so I have no idea. Even if you overpay Bill A it doesn't influence Bill B.

**Reply by Solverz-0** [reply]
Sorry, I must of not explained that very clearly, I am referring to the `Bills to pay` in the dashboard so it totals all the bills up.

**Reply by JC5** [reply]
Yes I know what you mean.

**Reply by Solverz-0** [reply]
Great, so the total in the dashboard if you over pay a bill gives the result like how I mentioned before, is this the desired behaviour?

**Reply by JC5** [reply]
It doesn't. I tried it and it doesn't do that. I'm not sure what you're doing but it doesn't use transaction data to calculate "left to pay".

**Reply by Solverz-0** [reply]
Ahh you are right, I just went through it again sorry!

I figured it out, must of been tired yesterday...
